api/jobs/handlers.py

```diff
@@ -271,3 +256,128 @@ def retry(self, _id):
 
         new_id = Queue.retry(j, force=True)
         return { "_id": new_id }
+
+class BatchHandler(base.RequestHandler):
+
+    @require_login
+    def get_all(self):
+        """
+        Get a list of batch jobs user has created.
+        Make a superuser request to see all batch jobs.
+        """
+
+        if self.superuser_request:
+            # Don't enforce permissions for superuser requests or drone requests
+            query = {}
+        else:
+            query = {'origin.id': self.uid}
+        return batch.get_all(query, {'proposed_inputs':0})
+
+    @require_login
+    def get(self, _id):
+        """
+        Get a batch job by id.
+        Use param jobs=true to replace job id list with job objects.
+        """
+
+        get_jobs = self.is_true('jobs')
+        batch_job = batch.get(_id, projection={'proposed_inputs':0}, get_jobs=get_jobs)
+        self._check_permission(batch_job)
+        return batch_job
+
+    @require_login
+    def post(self):
+        """
+        Create a batch job proposal, insert as 'pending'.
+        """
+
+        if self.superuser_request:
+            # Don't enforce permissions for superuser requests
+            user = None
+        else:
+            user = {'_id': self.uid, 'site': self.user_site}
+
+        payload = self.request.json
+        gear_name = payload.get('gear')
+        targets = payload.get('targets')
+        config_ = payload.get('config', {})
+
+        if not gear_name or not targets:
+            self.abort(400, 'A gear name and list of target containers is required.')
+        gear = get_gear_by_name(gear_name)
+        validate_gear_config(gear, config_)
+
+        container_ids = []
+        container_type = None
+
+        for t in targets:
+            if not container_type:
+                container_type = t.get('type')
+            else:
+                # Ensure all targets are of same type, may change in future
+                if container_type != t.get('type'):
```

Comment:
Currently assumes the target files will be at this level (ie, these containers are acquisitions not sessions). Will add target resolver for other hierarchy levels.